File: 1200/1352C/main.cpp
#include <bits/stdc++.h>
using namespace std;

int main() 
{
    int t;
    cin >> t;

    while (t--) 
    {
        long long n, k;
        cin >> n >> k;

        long long low = 1;
        long long high = k * n;

        while (low <= high) 
        {
            long long mid = low + (high - low) / 2;

            // Count of numbers <= mid that are not divisible by n
            if (mid - (mid / n) >= k)
                high = mid - 1;
            else
                low = mid + 1;
        }

        cout << low << "\n";
    }

    return 0;
}

File: 1200/1352C/main.java
import java.util.*;
public class main
{
    public static void main(String args[])
    {
        Scanner sc = new Scanner(System.in);
        
        int t = sc.nextInt();
        
        while(t-- > 0)
        {
            long n = sc.nextLong();
            long k = sc.nextLong();
            
            long low = 1;
            long high = k*n;
            
            while(low <= high)
            {
                long mid = low + (high-low)/2;
                
                if(mid-(mid/n) >= k)
                    high = mid-1;
                else
                    low = mid + 1;
            }
            
            System.out.println(low);
        }
    }
}